src/drafts/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,422 @@
+  useFocusTrap({containerRef: dialogRef, restoreFocusOnCleanUp: true, initialFocusRef: autoFocusedFooterButtonRef})
+
+  useOnEscapePress(
+    (event: KeyboardEvent) => {
+      onClose('escape')
+      event.preventDefault()
+    },
+    [onClose],
+  )
+
+  const header = (renderHeader ?? DefaultHeader)(defaultedProps)
+  const body = (renderBody ?? DefaultBody)(defaultedProps)
+  const footer = (renderFooter ?? DefaultFooter)(defaultedProps)
+
+  return (
+    <Portal>
+      <Backdrop ref={backdropRef}>
+        <StyledDialog
+          width={width}
+          height={height}
+          ref={dialogRef}
+          role={role}
+          aria-labelledby={dialogLabelId}
+          aria-describedby={dialogDescriptionId}
+          aria-modal
+          sx={sx}
+        >
+          {header}
+          {body}
+          {footer}
+        </StyledDialog>
+      </Backdrop>
+    </Portal>
+  )
+})
+_Dialog.displayName = 'Dialog'
+
+const Header = styled.div<SxProp>`
+  box-shadow: 0 1px 0 ${get('colors.border.default')};
+  padding: ${get('space.2')};
+  z-index: 1;
+  flex-shrink: 0;
+`
+
+// TODO: check if this is an appropriate use of `h1`
+const Title: React.FC<React.PropsWithChildren<React.ComponentPropsWithoutRef<'h1'> & SxProp>> = ({sx, ...props}) => (
+  <Text as="h1" margin="0" fontSize={1} sx={sx} {...props} />
+)
+
+// TODO: check if this is an appropriate use of `h2`
+const Subtitle: React.FC<React.PropsWithChildren<React.ComponentPropsWithoutRef<'h2'> & SxProp>> = ({sx, ...props}) => (
+  <Text as="h2" color="fg.muted" margin="0" fontWeight="normal" fontSize={0} sx={sx} {...props} />
+)
+
+const Body = styled.div<SxProp>`
+  flex-grow: 1;
+  overflow: auto;
+  padding: ${get('space.3')};
+
+  ${sx};
+`
+
+const Footer = styled.div<SxProp>`
+  box-shadow: 0 -1px 0 ${get('colors.border.default')};
+  padding: ${get('space.3')};
+  display: flex;
+  flex-flow: wrap;
+  justify-content: flex-end;
+  z-index: 1;
+  flex-shrink: 0;
+
+  button {
+    margin-left: ${get('space.1')};
+    &:first-child {
+      margin-left: 0;
+    }
+  }
+
+  ${sx};
+`
+
+const Buttons: React.FC<React.PropsWithChildren<{buttons: DialogButtonProps[]}>> = ({buttons}) => {
+  const autoFocusRef = useProvidedRefOrCreate<HTMLButtonElement>(buttons.find(button => button.autoFocus)?.ref)
+  let autoFocusCount = 0
+  const [hasRendered, setHasRendered] = useState(0)
+  useEffect(() => {
+    // hack to work around dialogs originating from other focus traps.
+    if (hasRendered === 1) {
+      autoFocusRef.current?.focus()
+    } else {
+      setHasRendered(hasRendered + 1)
+    }
+  }, [autoFocusRef, hasRendered])
+
+  return (
+    <>
+      {buttons.map((dialogButtonProps, index) => {
+        const {content, buttonType = 'normal', autoFocus = false, ...buttonProps} = dialogButtonProps
+        return (
+          <Button
+            key={index}
+            {...buttonProps}
+            variant={buttonType !== 'normal' ? buttonType : undefined}
```

Comment:
This also makes sense to me. `variant='default'` is equal to having no variant 👍🏻